src/ORM/FieldType/DBInt.php

```diff
@@ -40,14 +40,14 @@ public function setValue(mixed $value, null|array|ModelData $record = null, bool
         parent::setValue($value, $record, $markChanged);
         // Cast string ints if they're valid
         if (is_string($this->value) && preg_match('/^-?\d+$/', $this->value)) {
-            // Ensure we can cast to int and back without loss of precision
-            // if not, keep the original value which will fail validation later
-            $stringIntValue = (string) (int) $value;
-            if ($stringIntValue !== $value) {
-                $this->value = $value;
+            // Use filter_var to validate if the string fits in a PHP integer
+            // It will return false if it overflows i.e. is "9223372036854775808"
+            // meaning it's bigger than PHP_INT_MAX on 64-bit systems
```

Comment:
```suggestion
            // meaning it's bigger than PHP_INT_MAX
```
I think `FILTER_VALIDATE_INT` will return `false` for anything bigger than `PHP_INT_MAX` on a 32 bit system too, won't it?

Reply:
It would but there's a number "9223372036854775808" referenced above which relates to 64-bit system, this so makes sense in context

Reply:
Doesn't matter enough for ping pong, so leave as is